test/integration/service/loadbalancer_test.go

```diff
@@ -678,24 +720,16 @@ func Test_ServiceLoadBalancerIPMode(t *testing.T) {
 
 			controller, cloud, informer := newServiceController(t, client)
 			cloud.ExternalIP = net.ParseIPSloppy(tc.externalIP)
+			if tc.setIPMode {
+				cloud.BalancerIPMode = tc.expectedIngress.IPMode
```

Comment:
Setting the "inputs" from the expected value is weird; reading through the test cases it looks like it's just getting the ipmode from nowhere. It would make more sense to have `setIPMode *v1.LoadBalancerIPMode` instead of `setIPMode bool`?